### PR TITLE
Setting up global network. Fixes #64

### LINKED ISSUE #64: Polkadot treasury motion #29 | Milestone-1 intake review | Nice to have, is it possible to consider global `network` option ?
Nice to have, Do we need network selection at two different places ? is it possible to consider global `network` option ?

**Validation Setup**

| Component | Version |
| --- | --- |
| Release APK | litentry-governance_v1.0.1.apk |
| Android Version | 7.1.1 |
| Litentry Devnet | wss://18.140.130.138 |
| Parity Signer | v4.5.3-beta |

### PR DESCRIPTION
**Pull request by silva-fj**
With these changes users will be able to add an account only to the network they are connected.

Other relevant changes:
- New AccountsContext: With this new context we have the infrastructure in place to add the multi account support.
- Fixes some unnecessary re-renders related to accounts and the current network.
- NetworkSelectScreen was removed in favor of a new NetworkSelect modal to be consistent with the network switch from the Dashboard and to have single way to switch networks.
- Some types are fixed.

**Demo**

https://user-images.githubusercontent.com/8374946/122091601-42762a80-ce09-11eb-84cc-8aff5a8341cf.mp4

### CODE REVIEW

File: src/hoc/withAddAccount/index.tsx [hunk 1]
```diff
@@ -90,20 +80,20 @@ function withAddAccount<T>(Comp: React.ComponentType<T & InjectedPropTypes>) {
...
       }
-    }, [setAccount, state.address, state.network, state.step]);
+    }, [state.address, currentNetwork, state.step, addAccount]);
 
     const handleCancel = () => {
       modalRef.current?.close();
       dispatch({type: 'RESET'});
     };
 
     const confirmBtnDisabled = useMemo(() => {
```
Comment (ahoseinian) [inline src/hoc/withAddAccount/index.tsx:90]: Nit: I don't think `useMemo` is necessary here, it's just a boolean value.

File: src/screen/ApiLoadingScreen.tsx [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
 import ScreenNavigation from 'layout/ScreenNavigation';
 import NetworkItem from 'presentational/NetworkItem';
 import {NetworkContext} from 'context/NetworkContext';
-import {NavigationProp} from '@react-navigation/native';
 import {ChainApiContext} from 'context/ChainApiContext';
+import NetworkSelect from 'src/layout/NetworkSelect';
 
-export function ApiLoadingScreen({navigation}: {navigation: NavigationProp<any>}) {
+export function ApiLoadingScreen() {
+  const navigation = useNavigation();
```
Comment (ahoseinian) [inline src/screen/ApiLoadingScreen.tsx:13]: why are we not using the navigation prop?

**Review by ahoseinian — DISMISSED** [review]
LGTM

File: src/context/BalanceContext.tsx [hunk 3]
```diff
@@ -8,6 +7,7 @@ import {AccountInfo} from '@polkadot/types/interfaces';
 import ModalTitle from 'presentational/ModalTitle';
 import {NetworkContext} from './NetworkContext';
 import Balances from 'presentational/Balances';
+import {useAccounts} from 'src/context/AccountsContext';
```
Comment (geastwood) [inline src/context/BalanceContext.tsx:10]: no need the `src`, I guess your `vim` setup auto import like this.

**Review by geastwood — DISMISSED** [review]
Good work!

File: src/hoc/withAddAccount/index.tsx (hunk 1 above)
Comment (silva-fj) [reply src/hoc/withAddAccount/index.tsx:90]: true, it's not needed

File: src/context/BalanceContext.tsx (hunk 3 above)
Comment (silva-fj) [reply src/context/BalanceContext.tsx:10]: yeah, it's the auto import

File: src/screen/ApiLoadingScreen.tsx (hunk 2 above)
Comment (silva-fj) [reply src/screen/ApiLoadingScreen.tsx:13]: I was going to move this screen to a normal component but decided not to go that path. It works the same

File: ios/Podfile.lock [hunk 4]
```diff
@@ -12,7 +12,7 @@ PODS:
     - React-jsi (= 0.64.1)
```
Comment (silva-fj) [inline ios/Podfile.lock:12]: The changes here are coming from develop (dependencies updates from dependabot where merged without running `pod install`)

**Review by geastwood — APPROVED** [review]

**Pull request merged**